### PR TITLE
feat: resource upsert

### PR DESCRIPTION
**Pull request by arinda-arif**
Adding resource upsert API

Dependency: 
- [x] Merge proton [PR](https://github.com/goto/proton/pull/108)

### CODE REVIEW

File: client/cmd/resource/upload.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+func NewUploadCommand() *cobra.Command {
+	uploadCmd := &uploadCommand{
+		logger: logger.NewClientLogger(),
+	}
+
+	cmd := &cobra.Command{
+		Use:     "upload",
+		Short:   "Upload a resource to server",
+		Long:    heredoc.Doc(`Apply local changes to destination server which includes creating/updating resources`),
+		Example: "optimus resource upload -R [resource1,resource2] -n []",
```
Comment (sbchaos) [inline client/cmd/resource/upload.go:52]: We are providing single namespace in the command, are we accepting a list of namespaces here ?

Comment (arinda-arif) [reply client/cmd/resource/upload.go:52]: no, it should only a single namespace

File: core/resource/service/resource_service.go [hunk 2]
```diff
@@ -379,43 +436,44 @@ func (rs ResourceService) getResourcesToBatchUpdate(ctx context.Context, incomin
...
+	if resource.StatusIsSuccess(existing.Status()) && incoming.Equal(existing) {
+		_ = incoming.MarkSkipped()
+		rs.logger.Warn("resource [%s] is skipped because it has no changes", existing.FullName())
+		return nil
+	}
+
+	if resource.StatusForToCreate(existing.Status()) {
+		_ = incoming.MarkToCreate()
+	} else {
+		_ = incoming.MarkToUpdate()
```
Comment (arinda-arif) [inline core/resource/service/resource_service.go:471]: @sbchaos can you verify this change? as per my understanding, we should allow to update resources given any **_existing_** status. thus, making this change.

File: client/cmd/resource/upload.go (hunk 1 above)
Comment (arinda-arif) [reply client/cmd/resource/upload.go]: updated

File: core/resource/service/resource_service.go (hunk 2 above)
Comment (sbchaos) [reply core/resource/service/resource_service.go:471]: Sure, this should be safe to do

**Review by sbchaos — APPROVED** [review]

**Pull request merged**